### PR TITLE
chore: update to eslint 9

### PR DESCRIPTION
**Pull request by juliusmarminge**

### CODE REVIEW

File: packages/ui/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// FIXME: This kinda stinks...
+/// <reference types="../../tooling/eslint/types.d.ts" />
```
Comment (ixahmedxi) [inline packages/ui/eslint.config.js:2]: that does stink

Comment (juliusmarminge) [reply packages/ui/eslint.config.js]: fixed it

File: .vscode/settings.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "editor.formatOnSave": true,
   "eslint.rules.customizations": [{ "rule": "*", "severity": "warn" }],
+  "eslint.experimental.useFlatConfig": true,
```
Comment (juliusmarminge) [inline .vscode/settings.json:8]: looks like the new version is kinda close https://github.com/microsoft/vscode-eslint/releases

**Comment by RiccardoRomagnoli** [comment]
@juliusmarminge eslint-plugin-import@2.29.1 does not support eslint 9

**Pull request merged**